Bug1194767 - prefer slugid.nice() over slugid.v4()

petemoore opened the pull request:
Updated tutorials / docs as appropriate. Don't think I missed anything - but let me know if you think of something.

petemoore commented:
Also - please push to gh-pages branch after merging. Thanks!

jonasfj commented on learn/create-task/index.md:
```diff
@@ -179,8 +179,8 @@ a random slugid for use as `taskId`.
-// Generate a new taskId
-let taskId = slugid.v4();
+// Generate a new taskId (see [npm docs](https://www.npmjs.com/package/slugid))
+let taskId = slugid.nice();
```
No, here we should use `taskcluster.slugid()` that way we don't have to tell people to use nice...

Just tell them to use `taskcluster.slugid()`.

jonasfj commented:
We should advertise `taskcluster.slugid()`, so we don't have to explain the `.nice()` thing to anyone... It's too confusing for beginners...

jonasfj commented:
`r+` once moved to `taskcluster.slugid()`

petemoore commented:
So, like always, I felt compelled to do some more cleanup at the same time. Apologies for not putting the various cleanup in different PRs - I don't think it really matters though.

* [x] Removed old tools that have been superseded by tools on [tools.taskcluster.net](https://tools.taskcluster.net)
* [x] Updated tools page to have a thumbnail of [tools.taskcluster.net](https://tools.taskcluster.net) which links to the site
* [x] @jhford Removed page about legacy aws-provisioner
* [x] Updated reference in generic worker docs to the old provisioner, to be the new one
* [x] Updated tutorials to use `taskcluster.slugid()` method rather than using slugid library directly
* [x] "Rebrowserified" `taskcluster-client` to pick up latest changes, including the `slugid()` method
* [x] Deleted now-unused browserified `slugid` module, as interactive tutorials no longer need it, and legacy task creator/task inspector/task graph inspector tools have been removed

jonasfj commented:
Looked reasonable... merging...
